components/suggestion/modal_suggestion_list.jsx

```diff
@@ -0,0 +1,172 @@
+            return;
+        }
+
+        if (prevProps.open !== this.state.open ||
+            prevProps.cleared !== this.state.cleared ||
+            prevState.scroll !== this.state.scroll ||
+            prevState.modalBounds.top !== this.state.modalBounds.top ||
+            prevState.modalBounds.bottom !== this.state.modalBounds.bottom) {
+            const newInputBounds = this.updateInputBounds();
+            this.updateLocation(newInputBounds);
+
+            if (this.container.current) {
+                const modalBodyRect = getClosestParent(this.container.current, '.modal-body').getBoundingClientRect();
+                if ((newInputBounds.bottom < modalBodyRect.top) || (newInputBounds.top > modalBodyRect.bottom)) {
+                    this.props.onLoseVisibility();
+                    return;
+                }
+            }
+
+            this.updateModalBounds();
+        }
+    }
+
+    getChildHeight = () => {
+        if (!this.container.current) {
+            return 0;
+        }
+
+        const listElement = this.suggestionList.current.getContent()[0];
+        if (!listElement) {
+            return 0;
+        }
+
+        return listElement.getBoundingClientRect().height;
+    }
+
+    updateInputBounds = () => {
+        const inputBounds = this.props.calculateInputRect();
+        if (inputBounds.top !== this.state.inputBounds.top ||
+            inputBounds.bottom !== this.state.inputBounds.bottom ||
+            inputBounds.width !== this.state.inputBounds.width) {
+            this.setState({inputBounds});
+        }
+        return inputBounds;
+    }
+
+    updateLocation = (newInputBounds) => {
+        let inputBounds = newInputBounds;
+        if (!newInputBounds) {
+            inputBounds = this.state.inputBounds;
+        }
+
+        if (!this.container.current) {
+            return;
+        }
+
+        this.latestHeight = this.getChildHeight();
+
+        let newLocation = this.props.location;
+        if (window.innerHeight < inputBounds.bottom + this.latestHeight) {
+            newLocation = 'top';
+        }
+        if (inputBounds.top - this.latestHeight < 0) {
+            newLocation = 'bottom';
+        }
+
+        if (this.state.location !== newLocation) {
+            this.setState({location: newLocation});
+        }
+    }
+
+    updateModalBounds = () => {
+        if (!this.container.current) {
+            return;
+        }
+
+        const modalContainer = getClosestParent(this.container.current, '.modal-content');
+        const modalBounds = modalContainer.getBoundingClientRect();
+
+        if (this.state.modalBounds.top !== modalBounds.top || this.state.modalBounds.bottom !== modalBounds.bottom) {
+            this.setState({modalBounds: {top: modalBounds.top, bottom: modalBounds.bottom}});
+        }
+    }
+
+    render() {
+        const {
+            ...props
+        } = this.props;
+
+        Reflect.deleteProperty(props, 'onLoseVisibility');
```

Comment:
The problem in doing it your way is that we would declare two variables that will never be used, and if I am not mistaken, will generate some lint errors.

Reply:
Yeah, I'd suggest the destructuring method if we were using those props, but since we aren't, I think `Reflect.deleteProperty` is the right way to go here

Reply:
@larkox 
Ahh fair enough, how about this instead?


```
const {
	onLoseVisibility: _,
	calculateinputRect: _,
	...otherProps
} = this.props;
```

The reason i prefer this is because everything is inside one single block as opposed to the current approach of having two operations, one is destructing and then removing items.

Reply:
That approach yields the following lint error:
`'_' is assigned a value but never used`

The current approach is being used in several parts of the code, and I would consider it a standard in the code right now. Nevertheless, if you come up with any other idea, I am more than happy to try.

Reply:
thanks  for giving it a try @larkox! :)